Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/features/skylanders/ui/SkylanderSlotAdapter.java

```diff
@@ -0,0 +1,155 @@
+    {
+      super(binding.getRoot());
+      this.binding = binding;
+    }
+  }
+
+  private final List<SkylanderSlot> mSlots;
+  private final EmulationActivity mContext;
+
+  private DialogCreateSkylanderBinding mBinding;
+
+  public SkylanderSlotAdapter(List<SkylanderSlot> slots, EmulationActivity context)
+  {
+    mSlots = slots;
+    mContext = context;
+  }
+
+  @NonNull
+  @Override
+  public SkylanderSlotAdapter.ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType)
+  {
+    LayoutInflater inflater = LayoutInflater.from(parent.getContext());
+    ListItemSkylanderSlotBinding binding =
+            ListItemSkylanderSlotBinding.inflate(inflater, parent, false);
+    return new ViewHolder(binding);
+  }
+
+  @Override
+  public void onBindViewHolder(@NonNull ViewHolder holder, int position)
+  {
+    SkylanderSlot slot = mSlots.get(position);
+    holder.binding.textSkylanderName.setText(slot.getLabel());
+
+    holder.binding.buttonClearSkylander.setOnClickListener(
+            v ->
+            {
+              SkylanderConfig.removeSkylander(slot.getmSlotNum());
+              mContext.clearSkylander(slot.getmSlotNum());
+            });
+
+
+    holder.binding.buttonLoadSkylander.setOnClickListener(v ->
+    {
+      Intent loadSkylander = new Intent(Intent.ACTION_OPEN_DOCUMENT);
+      loadSkylander.addCategory(Intent.CATEGORY_OPENABLE);
+      loadSkylander.setType("*/*");
+      loadSkylander.putExtra("Slot", slot.getmSlotNum());
+      mContext.startActivityForResult(loadSkylander, EmulationActivity.REQUEST_SKYLANDER_FILE);
+    });
+
+    LayoutInflater inflater = LayoutInflater.from(mContext);
+    mBinding = DialogCreateSkylanderBinding.inflate(inflater);
+
+    List<String> skylanderNames = new ArrayList<>(Skylander.REVERSE_LIST_SKYLANDERS.keySet());
+    skylanderNames.sort(String::compareToIgnoreCase);
+
+    mBinding.skylanderDropdown.setAdapter(
+            new ArrayAdapter<>(mContext, R.layout.support_simple_spinner_dropdown_item,
+                    skylanderNames));
+    mBinding.skylanderDropdown.setOnItemClickListener(this);
+
+    holder.binding.buttonCreateSkylander.setOnClickListener(v ->
+    {
+      if (mBinding.getRoot().getParent() != null)
+      {
+        ((ViewGroup) mBinding.getRoot().getParent()).removeAllViews();
+
+      }
+      new MaterialAlertDialogBuilder(mContext)
+              .setTitle(R.string.create_skylander)
+              .setView(mBinding.getRoot())
+              .setPositiveButton(R.string.create_skylander, (dialog, which) ->
+              {
+                Intent createSkylander = new Intent(Intent.ACTION_CREATE_DOCUMENT);
+                createSkylander.addCategory(Intent.CATEGORY_OPENABLE);
+                createSkylander.setType("*/*");
+                Integer id = Integer.valueOf(mBinding.skylanderId.getText().toString());
+                Integer var = Integer.valueOf(mBinding.skylanderVar.getText().toString());

```

Comment:
Use parseInt instead
```suggestion
                Integer id = Integer.parseInt(mBinding.skylanderId.getText().toString());
                Integer var = Integer.parseInt(mBinding.skylanderVar.getText().toString());

```